prov/sm2/src/sm2_progress.c

```diff
@@ -281,12 +428,103 @@ static int sm2_alloc_xfer_entry_ctx(struct sm2_ep *ep,
 static int sm2_progress_recv_msg(struct sm2_ep *ep,
 				 struct sm2_xfer_entry *xfer_entry)
 {
+	struct sm2_av *av =
+		container_of(ep->util_ep.av, struct sm2_av, util_av);
+	struct sm2_mmap *map = &av->mmap;
 	struct fid_peer_srx *peer_srx = sm2_get_peer_srx(ep);
 	struct fi_peer_rx_entry *rx_entry;
+	struct sm2_cma_data *cma_data;
 	struct sm2_av *sm2_av;
 	fi_addr_t addr;
+	uint64_t comp_flags;
 	int ret;
 
+	/* TODO - Switch on protocol before switching on op to avoid messy
+	 * checks like this */
+
+	/* We don't want to look for rx entries for xfer entries with
+	 * FI_SM2_CMA_HOST_TO_DEV and FI_SM2_CMA_HOST_TO_DEV_ACK flags*/
+	if (xfer_entry->hdr.proto == sm2_proto_cma) {
+		if (xfer_entry->hdr.proto_flags & FI_SM2_CMA_HOST_TO_DEV) {
+			/* Sender received IPC handle from receiver and needs to
+			 * do a device memcpy, so skip rx entry lookup */
+			ret = sm2_cma_hmem_memcpy(ep, xfer_entry);
+			if (ret) {
+				FI_WARN(&sm2_prov, FI_LOG_EP_CTRL,
+					"Device memcpy in host to device IPC "
+					"protocol "
```

Comment:
Go through your strings and combine the lines... if it gets split and re-split... sometimes clang-format doesn't handle it gracefully

Reply:
this was the only wonky string